src/sensr_client.cpp

```diff
@@ -1,13 +1,20 @@
 #include "sensr_client.h"
-#include "websocket_endpoint.h"
+#include "websocket/websocket_endpoint.h"
+#include "websocket/websocket_secure_endpoint.h"
 #include "sensr_message_listener.h"
 #include <algorithm>
 
 namespace sensr
 {
-  Client::Client(const std::string& address) : address_(address) {
-    output_endpoint_.reset(new WebSocketEndPoint());
-    point_endpoint_.reset(new WebSocketEndPoint());
+  Client::Client(const std::string& address, const std::string& cert_path) 
+      : address_(address), use_ssl_(!cert_path.empty()) {
+    if (use_ssl_) {
+      output_endpoint_.reset(new WebSocketSecureEndPoint(cert_path));
+      point_endpoint_.reset(new WebSocketSecureEndPoint(cert_path));
+    } else {
+      output_endpoint_.reset(new WebSocketEndPoint());
+      point_endpoint_.reset(new WebSocketEndPoint());
```

Comment:
can we use `std::make_unique` here instead?

Reply:
I believe this has to support C++11, which didn't include `std::make_unique`. I think I made the same exact comment in a previous PR :laughing:

Reply:
Ahh :joy: Good to know that

Reply:
Yes, SDK use c++11 yet to provide better compatibility.